src/crypto/openssl/cose_signer.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/openssl/cose_signer.h"
+
+#include "ccf/ds/logger.h"
+
+#include <openssl/evp.h>
+#include <t_cose/t_cose_sign1_sign.h>
+
+namespace ccf::crypto
+{
+  using namespace OpenSSL;
+
+  COSESigner_OpenSSL::COSESigner_OpenSSL(const Pem& key) :
+    key_pair(std::make_shared<KeyPair_OpenSSL>(key))
+  {}
+
+  bool COSESigner_OpenSSL::sign(
+    std::span<uint8_t> payload, std::vector<uint8_t>& cose_envelope) const
+  {
+    struct t_cose_sign1_sign_ctx sign_ctx;
+    t_cose_sign1_sign_init(&sign_ctx, 0, T_COSE_ALGORITHM_ES256);
+
+    EVP_PKEY* evp_key = *key_pair;
+
+    t_cose_key signing_key;
+    signing_key.crypto_lib = T_COSE_CRYPTO_LIB_OPENSSL;
+    signing_key.k.key_ptr = evp_key;
+
+    t_cose_sign1_set_signing_key(&sign_ctx, signing_key, NULL_Q_USEFUL_BUF_C);
+
+    // Neither documentation or code example explain how much memory should be
+    // reserved. Encoding will fail gracefully if the buffer was too small.
+    const size_t payload_buffer_size{payload.size() * 2 + 128};
```

Comment:
128 seems like a magic number may be define it as 
    // constexpr size_t PAYLOAD_BUFFER_EXTRA_SIZE = 128;

Reply:
This is now deleted